lib/elastic_apm.rb

```diff
@@ -167,15 +167,17 @@ def start_span(
       action: nil,
       context: nil,
       include_stacktrace: true,
-      trace_context: nil
+      trace_context: nil,
+      parent_transaction: nil
```

Comment:
Why `parent_transaction` and not just `transaction`? The span is always related to only _one_ transaction and it'll always be its parent.

Reply:
It's to keep the naming consistent with the argument in the `instrumenter` method. In that method, I need a way to refer to a current_transaction, variable transaction, and argument transaction. [These lines](https://github.com/elastic/apm-agent-ruby/pull/571/files#diff-b907effb9782fac8cec4233c342822a6R149) could change so that I don't need to use a `transaction` variable.
```
transaction = parent_transaction || current_transaction
return unless transaction
return unless transaction.sampled?	
```

to

```
return unless(parent_transaction || current_transaction)
return unless(parent_transaction || current_transaction).sampled?	
```

Reply:
Do you have any other naming suggestions for how to distinguish between the argument transaction and current transaction?

Reply:
I don't have any good suggestions unless you want me to dive further in?

My point was more from the perspective of the user. This is the top-most, outward facing API. And for the user, `parent_` is given – at worst confusing; eg _why wouldn't it be?_

I think we should start from there and then dance our way around juggling the different internal complexities afterwards if we have to. Makes sense?

Reply:
ok, I see your point. I'll make the argument `transaction` and see how I can handle it in the `instrumenter`.

Reply:
I changed the argument to `transaction`